test/integration/acceptance/fake_peer_example_test.cpp

```diff
@@ -107,3 +110,174 @@ TEST_F(FakePeerExampleFixture,
+      block.signAndAddSignature(peer->getKeypair());
+    }
+    return std::move(block);
+  };
+
+  // Create the malicious fork of the ledger:
+  auto bad_block_storage =
+      std::make_shared<fake_peer::BlockStorage>(*valid_block_storage);
+  bad_block_storage->storeBlock(std::make_shared<shared_model::proto::Block>(
+      sign_block_by_peers(
+          proto::BlockBuilder()
+              .height(valid_block_storage->getTopBlock()->height() + 1)
+              .prevHash(valid_block_storage->getTopBlock()->hash())
+              .createdTime(iroha::time::now())
+              .transactions(
+                  std::initializer_list<shared_model::proto::Transaction>{
+                      complete(
+                          baseTx(kAdminId).transferAsset(
+                              kAdminId, kUserId, kAssetId, "bad_tx3", "300.0"),
+                          kAdminKeypair)})
+              .build(),
+          bad_fake_peers)
+          .finish()));
+  for (auto &bad_fake_peer : bad_fake_peers) {
+    bad_fake_peer->setBlockStorage(bad_block_storage);
+  }
+
+  // Extend the valid ledger:
+  valid_block_storage->storeBlock(std::make_shared<shared_model::proto::Block>(
+      sign_block_by_peers(
+          proto::BlockBuilder()
+              .height(valid_block_storage->getTopBlock()->height() + 1)
+              .prevHash(valid_block_storage->getTopBlock()->hash())
+              .createdTime(iroha::time::now())
+              .transactions(
+                  std::initializer_list<shared_model::proto::Transaction>{
+                      complete(
+                          baseTx(kAdminId).transferAsset(
+                              kAdminId, kUserId, kAssetId, "valid_tx3", "3.0"),
+                          kAdminKeypair)})
+              .build(),
+          good_fake_peers)
+          .finish()));
+  for (auto &good_fake_peer : good_fake_peers) {
+    good_fake_peer->setBlockStorage(valid_block_storage);
+  }
+
+  // Create the new block that the good peers are about to commit now.
+  auto new_valid_block = std::make_shared<shared_model::proto::Block>(
+      sign_block_by_peers(
+          proto::BlockBuilder()
+              .height(valid_block_storage->getTopBlock()->height() + 1)
+              .prevHash(valid_block_storage->getTopBlock()->hash())
+              .createdTime(iroha::time::now())
+              .transactions(
+                  std::initializer_list<shared_model::proto::Transaction>{
+                      complete(
+                          baseTx(kAdminId).transferAsset(
+                              kAdminId, kUserId, kAssetId, "valid_tx4", "4.0"),
+                          kAdminKeypair)})
+              .build()
+              .signAndAddSignature(rantipole_peer->getKeypair()),
+          good_fake_peers)
+          .finish());
+
+  // From now the itf peer is considered reachable from the rest network. //
+
+  // Suppose the rantipole peer created a valid Commit message for the tip of
```

Comment:
rantipole peer 😱

Reply:
yep

Reply:
I don't really think it's a good name, but meh, fix is optional